src/test/java/com/browserstack/steps/SetupSteps.java

```diff
@@ -0,0 +1,127 @@
+            stepData.url = URL;
+        } else if (StringUtils.isNoneEmpty(System.getProperty("env")) && System.getProperty("env").equalsIgnoreCase("docker")) {
+            caps.setBrowserName("chrome");
+            caps.setCapability(CapabilityType.ACCEPT_SSL_CERTS, true);
+            stepData.webDriver = new RemoteWebDriver(new URL("http://localhost:4444/wd/hub"), caps);
+            stepData.url = URL;
+        } else {
+            if(System.getenv("caps")!= null) {
+                config = (JSONObject) parser.parse(System.getenv("caps"));
+            } else {
+                config = (JSONObject) parser.parse(new FileReader("src/test/resources/config/caps.json"));
+            }
+            if (System.getProperty("parallel") != null) {
+                capabilityObject = ParallelTest.threadLocalValue.get();
+            } else {
+                JSONObject singleCapabilityJson = (JSONObject) ((JSONObject) config.get("tests")).get("single");
+                JSONArray environments = (JSONArray)singleCapabilityJson.get("env_caps");
+                capabilityObject = Utility.getCombinedCapability((Map<String, String>) environments.get(0),config,singleCapabilityJson);
+            }
+
+            Map<String, String> commonCapabilities = (Map<String, String>) capabilityObject.get("capabilities");
+            Iterator it = commonCapabilities.entrySet().iterator();
+            while (it.hasNext()) {
+                Map.Entry pair = (Map.Entry) it.next();
+                if (caps.getCapability(pair.getKey().toString()) == null) {
+                    caps.setCapability(pair.getKey().toString(), pair.getValue().toString());
+                }
+            }
+            caps.setCapability("name", scenario.getName());
+
+            String username = System.getenv("BROWSERSTACK_USERNAME");
+            if (username == null) {
+                username = (String) capabilityObject.get("user");
+            }
+            String accessKey = System.getenv("BROWSERSTACK_ACCESS_KEY");
+            if (accessKey == null) {
+                accessKey = (String) capabilityObject.get("key");
+            }
+            stepData.url = (String) capabilityObject.get("application_endpoint");
+            if (caps.getCapability("browserstack.local") != null && caps.getCapability("browserstack.local").equals("true")) {
+                String localIdentifier = RandomStringUtils.randomAlphabetic(8);
+                caps.setCapability("browserstack.localIdentifier", localIdentifier);
+                local = new Local();
+                Map<String, String> options = Utility.getLocalOptions(config);
+                options.put("key", accessKey);
+                options.put("localIdentifier", localIdentifier);
+                local.start(options);
+            }
+            System.out.println(caps.toString());
+            String URL = String.format("https://%s:%s@hub.browserstack.com/wd/hub", username, accessKey);
```

Comment:
what do you think of passing username / access key in caps['browserstack.user'] and caps['browserstack.access_key']? the URL is not encrypted in HTTPS but the data is.

Reply:
done

Reply:
done